gnocchi/storage/redis.py

```diff
@@ -97,35 +97,44 @@ def _get_or_create_unaggregated_timeseries(self, metrics, version=3):
         }
         return ts
 
-    def _list_split_keys(self, metric, aggregations, version=3):
-        key = self._metric_key(metric)
+    def _list_split_keys(self, metrics_and_aggregations, version=3):
         pipe = self._client.pipeline(transaction=False)
-        pipe.exists(key)
-        for aggregation in aggregations:
-            self._scripts["list_split_keys"](
-                keys=[key], args=[self._aggregated_field_for_split(
-                    aggregation.method, "*",
-                    version, aggregation.granularity)],
-                client=pipe,
-            )
+        # Keep an ordered list of metrics
+        metrics = list(metrics_and_aggregations.keys())
+        for metric in metrics:
+            key = self._metric_key(metric)
+            pipe.exists(key)
+            aggregations = metrics_and_aggregations[metric]
+            for aggregation in aggregations:
+                self._scripts["list_split_keys"](
+                    keys=[key], args=[self._aggregated_field_for_split(
+                        aggregation.method, "*",
+                        version, aggregation.granularity)],
+                    client=pipe,
+                )
         results = pipe.execute()
-        metric_exists_p = results.pop(0)
-        if not metric_exists_p:
-            raise storage.MetricDoesNotExist(metric)
-        keys = {}
-        for aggregation, k in six.moves.zip(aggregations, results):
-            if not k:
-                keys[aggregation] = set()
-                continue
-            timestamps, methods, granularities = list(zip(*k))
-            timestamps = utils.to_timestamps(timestamps)
-            granularities = map(utils.to_timespan, granularities)
-            keys[aggregation] = {
-                carbonara.SplitKey(timestamp,
-                                   sampling=granularity)
-                for timestamp, granularity
-                in six.moves.zip(timestamps, granularities)
-            }
+        keys = collections.defaultdict(dict)
+        for metric in metrics:
+            metric_exists_p = results.pop(0)
+            if not metric_exists_p:
+                raise storage.MetricDoesNotExist(metric)
+            aggregations = metrics_and_aggregations[metric]
+            keys_for_aggregations = results[:len(aggregations)]
+            del results[:len(aggregations)]
+            for aggregation, k in six.moves.zip(
+                    aggregations, keys_for_aggregations):
+                if not k:
+                    keys[metric][aggregation] = set()
+                    continue
+                timestamps, methods, granularities = list(zip(*k))
```

Comment:
what's `methods`?

can't we expand this when we define for loop? like 
`for aggregation, (timestamps, granularities) in six.moves.zip(aggregations, keys_for_aggregations):`

Reply:
As a note, this is just how the code was written previously, not new code.

> what's `methods`?

The aggregation method as a string parsed out from the Redis key name by the Lua script.

> can't we expand this when we define for loop? like
for aggregation, (timestamps, granularities) in six.moves.zip(aggregations, keys_for_aggregations)

No, I think you misread the code.

Reply:
word. i was reviewing in unified view and didn't notice the original.